.github/workflows/test.yml

```diff
@@ -0,0 +1,79 @@
+name: "Test"
+on:
+    push:
+        branches:
+            - master
+        tags:
+            - "*"
+    pull_request:
+        types: [opened, reopened, synchronize]
+jobs:
+    unit:
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                node: [8, 10, 11, 12]
+        env:
+            TEST: "unit"
+        steps:
+            - uses: actions/setup-node@v1
+              with:
+                  node-version: ${{ matrix.node }}
+            - uses: actions/checkout@v1
+            - run: bash ./scripts/install.sh
+            - run: bash ./scripts/ci.sh
+    build_and_lint:
+        runs-on: ${{ matrix.os }}
+        env:
+            TEST: "build_and_lint"
+        steps:
+            - uses: actions/setup-node@v1
+              with:
+                  node-version: 10
+            - uses: actions/checkout@v1
+            - run: bash ./scripts/install.sh
+            - run: bash ./scripts/ci.sh
+    e2e:
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                testCmd:
+                    [
+                        "unit_and_e2e_clients",
+                        "e2e_ganache",
+                        "e2e_mosaic",
+                        "e2e_ens",
+                        "e2e_browsers",
+                    ]
+        env:
+            TEST: ${{ matrix.testCmd }}
+            CI: true
+            DISPLAY: :99.0
+            PUPPETEER_SKIP_CHROMIUM_DOWNLOAD: true
+        steps:
+            - uses: actions/setup-node@v1
+              with:
+                  node-version: 12
+            - uses: actions/checkout@v1
+            - run: bash ./scripts/install.sh
+
+            - run: bash ./scripts/ci.sh
+              continue-on-error: ${{ matrix.testCmd == 'e2e_ganache' || matrix.testCmd == 'e2e_mosiac' }}
+
+            - name: Send coverage reports to Coveralls
+              if: env.TEST == 'unit_and_e2e_clients'
+              uses: coverallsapp/github-action@master
+              with:
+                  github-token: ${{ secrets.GITHUB_TOKEN }}
+    e2e_windows:
+        runs-on: windows-latest
+        env:
+            TEST: "e2e_windows"
+            CI: true
+        steps:
+            - uses: actions/setup-node@v1
+              with:
+                  node-version: 12
+            - uses: actions/checkout@v1
+            - run: bash ./scripts/install.sh
+            - run: bash ./scripts/ci.sh
```

Comment:
This might need to be `continue-on-error: true` too (at least temporarily). `e2e_windows` is a verdaccio/lerna job and lerna won't publish when git is in a detached head state. 

For PRs made directly to the repo by maintainers the relevant branch can just be checked out...

https://github.com/ethereum/web3.js/blob/1149c418e8c0a85270138395769fc31927bf9c25/scripts/e2e.npm.publish.sh#L56

...but for PRs from forks believe we'd need to create that branch and explicitly pull from their remote. That logic's missing atm.

Reply:
ok thanks for the note, will add for now